armi/reactor/converters/uniformMesh.py

```diff
@@ -244,13 +236,33 @@ def _buildAllUniformAssemblies(self):
             src = sourceAssem.spatialLocator
             newLoc = self.convReactor.core.spatialGrid[src.i, src.j, 0]
             self.convReactor.core.add(newAssem, newLoc)
+        self.plotConvertedReactor()
 
     def plotConvertedReactor(self):
-        assemsToPlot = self.convReactor.core[:12]
-        for plotNum, assemBatch in enumerate(iterables.chunk(assemsToPlot, 6), start=1):
-            assemPlotName = f"{self.convReactor.core.name}AssemblyTypes{plotNum}.png"
+        bpAssems = list(self.convReactor.blueprints.assemblies.values())
+        assemsToPlot = []
+        for bpAssem in bpAssems:
+            coreAssems = self.convReactor.core.getAssemblies(bpAssem.p.flags)
+            if not coreAssems:
+                continue
+            assemsToPlot.append(coreAssems[0])
+
+        # Obtain the plot numbering based on the existing files so that existing plots
+        # are not overwritten.
+        start = 0
+        existingFiles = glob.glob(
+            f"{self.convReactor.core.name}AssemblyTypes" + "*" + ".png"
+        )
+        for f in existingFiles:
+            newStart = int(re.search(r"\d+", f).group())
```

Comment:
Sorry, I can't tell...  What is the regex here for?

At the very least, we need a good comment describing what this regex is doing, because if I can't tell, neither can others.